app/src/sharedTest/java/org/oppia/app/parser/StringToRatioParserTest.kt

```diff
@@ -0,0 +1,131 @@
+    stringToRatioParser = StringToRatioParser()
+  }
+
+  @Test
+  fun testParser_realtimeError_answerWithAlphabets_returnInvalidCharsError() {
+    val error =
+      stringToRatioParser.getRealTimeAnswerError("abc").getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo(
+      "Please write a ratio that consists of digits separated by colons (e.g. 1:2 or 1:2:3)."
+    )
+  }
+
+  @Test
+  fun testParser_realtimeError_answerWithTwoAdjacentColons_returnInvalidColonsError() {
+    val error = stringToRatioParser.getRealTimeAnswerError("1::2")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Your answer has two colons (:) next to each other.")
+  }
+
+  @Test
+  fun testParser_realtimeError_answerWithCorrectRatio_returnValid() {
+    val error = stringToRatioParser.getRealTimeAnswerError("1:2:3")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo(null)
+  }
+
+  @Test
+  fun testParser_submitTimeError_returnInvalidSizeError() {
+    val error = stringToRatioParser.getSubmitTimeError("1:2:3:4", numberOfTerms = 5)
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Number of terms is less than required terms.")
+  }
+
+  @Test
+  fun testParser_submitTimeError_answerWithOneExtraColon_returnInvalidFormatError() {
+    val error =
+      stringToRatioParser.getSubmitTimeError("1:2:3:", 3).getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid ratio (e.g. 1:2 or 1:2:3).")
+  }
+
+  @Test
+  fun testParser_submitTimeError_answerWithZeroComponent_returnIncludesZero() {
+    val error =
+      stringToRatioParser.getSubmitTimeError("1:2:0", 3).getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Ratios cannot have 0 as a element.")
+  }
+
+  @Test
+  fun testParser_submitTimeError_returnValid() {
+    val error = stringToRatioParser.getSubmitTimeError("1:2:3:4", 4)
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo(null)
+  }
+
+  @Test
+  fun testParser_parseRatioOrNull_returnRatioExpression() {
+    val parsedRatio = stringToRatioParser.parseRatioOrNull("1:2:3:4")
+    val constructedRatio = createRatio(listOf(1, 2, 3, 4))
+    assertThat(parsedRatio).isEqualTo(constructedRatio)
+  }
+
+  @Test
+  fun testParser_parseRatioOrNull_returnNull() {
+    val parsedRatio = stringToRatioParser.parseRatioOrNull("1:2:3:4:")
+    assertThat(parsedRatio).isEqualTo(null)
+  }
+
+  @Test
+  fun testParser_parseRatioOrThrow_ratioWithWhiteSpaces_returnRatioExpression() {
+    val parsedRatio = stringToRatioParser.parseRatioOrThrow("1   :   2   : 3: 4")
+    val constructedRatio = createRatio(listOf(1, 2, 3, 4))
+    assertThat(parsedRatio).isEqualTo(constructedRatio)
+  }
+
+  @Test
+  fun testParser_parseRatioOrThrow_ratioWithInvalidRatio_throwsException() {
+
+    val exception = assertThrows(IllegalArgumentException::class) {
```

Comment:
Nit: please remove empty newline above for consistency with other tests.

Reply:
done